datumaro/plugins/open_images_format.py

```diff
@@ -0,0 +1,155 @@
+            for class_description in class_description_reader:
+                label_categories.add(class_description['LabelName'])
+
+        self._categories[AnnotationType.label] = label_categories
+
+        self._load_label_category_parents()
+
+    def _load_label_category_parents(self):
+        label_categories = self._categories[AnnotationType.label]
+
+        hierarchy_path = osp.join(
+            self._dataset_dir, _OpenImagesPath.ANNOTATIONS_DIR, 'bbox_labels_600_hierarchy.json')
+
+        with open(hierarchy_path, 'rb') as hierarchy_file:
+            root_node = json.load(hierarchy_file)
+
+        def set_parents_from_node(node, category):
+            for child_node in node.get('Subcategory', []):
+                _, child_category = label_categories.find(child_node['LabelName'])
+
+                if category is not None and child_category is not None:
+                    child_category.parent = category.name
+
+                set_parents_from_node(child_node, child_category)
+
+        _, root_category = label_categories.find(root_node['LabelName'])
+        set_parents_from_node(root_node, root_category)
+
+    def _load_items(self):
+        items_by_id = {}
+
+        def load_from(path):
+            with self._open_csv_annotation(path) as image_reader:
+                for image_description in image_reader:
+                    image_id = image_description['ImageID']
+                    assert _RE_IMAGE_ID.fullmatch(image_id)
+                    assert image_id not in items_by_id
+
+                    subset = image_description['Subset']
+                    assert subset in {'test', 'train', 'validation'}
+
+                    items_by_id[image_id] = DatasetItem(
+                        id=image_id,
+                        image=osp.join(self._dataset_dir, f'images/{subset}/{image_id}.jpg'),
+                        subset=subset,
+                    )
+
+        # It's preferable to load the combined image description file,
+        # because it contains descriptions for training images without human-annotated labels
+        # (the file specific to the training set doesn't).
+        # However, if it's missing, we'll try loading subset-specific files instead, so that
+        # this exporter can be used on individual subsets of the dataset.
+        try:
+            load_from(_OpenImagesPath.FULL_IMAGE_DESCRIPTION_NAME)
+        except FileNotFoundError:
+            for pattern in _OpenImagesPath.SUBSET_IMAGE_DESCRIPTION_PATTERNS:
+                for path in self._glob_annotations(pattern):
+                    load_from(path)
+
+        self._items.extend(items_by_id.values())
+
+        self._load_labels(items_by_id)
+
+    def _load_labels(self, items_by_id):
+        label_categories = self._categories[AnnotationType.label]
+
+        # TODO: implement reading of machine-annotated labels
+
+        for label_path in self._glob_annotations('*-human-imagelabels.csv'):
```

Comment:
Consider `os.listdir` and `glob.iglob`.

Reply:
`iglob` I can understand, but how would `listdir` help me here?

Reply:
Well, I'm not sure if this is what you meant, but I redid the implementation of `_glob_annotations` so that the directory is only listed once.